doc/dts/device_tree.rst

```diff
@@ -0,0 +1,208 @@
+.. _device-tree:
+
+Device Tree in Zephyr
+########################
+
+Introduction to Device Tree
+***************************
+
+Device tree is a way of describing hardware and configuration information
```

Comment:
Could use some more details for folks that don't know much about device tree.  Maybe replace this first paragraph with:

The device tree is a data structure for dynamically describing hardware.  Rather than hard coding every detail of a board's hardware into the operating system, the hardware is described in the device tree data structure and passed to the operating system at boot time as a compiled device tree "blob".  This allows the same compiled Linux kernel to support different hardware configurations within an architecture family (e.g., ARM, x86, PowerPC) and moves a signiﬁcant part of the hardware description into a data structure that is no longer part of the kernel binary itself.